zone/zonedb.cpp

```diff
@@ -997,27 +998,33 @@ void ZoneDatabase::LoadCharacterTribute(Client* c){
 	}
 }
 
-bool ZoneDatabase::LoadCharacterPotions(uint32 character_id, PlayerProfile_Struct *pp)
+bool ZoneDatabase::LoadCharacterPotionBelt(uint32 character_id, PlayerProfile_Struct *pp)
 {
-	std::string query =
-	    StringFormat("SELECT `potion_id`, `item_id`, `icon` FROM `character_potionbelt` WHERE `id` = %u LIMIT %u",
-		character_id, EQ::profile::POTION_BELT_SIZE);
-	auto results = database.QueryDatabase(query);
-	int i = 0;
-	for (i = 0; i < EQ::profile::POTION_BELT_SIZE; i++) {
+	const auto& l = CharacterPotionbeltRepository::GetWhere(
+		database,
+		fmt::format(
+			"`id` = {} LIMIT {}",
+			character_id,
+			EQ::profile::POTION_BELT_SIZE
+		)
+	);
+
+	for (int i = 0; i < EQ::profile::POTION_BELT_SIZE; i++) { // Initialize Potion Belt
 		pp->potionbelt.Items[i].Icon = 0;
-		pp->potionbelt.Items[i].ID = 0;
+		pp->potionbelt.Items[i].ID   = 0;
 		pp->potionbelt.Items[i].Name[0] = '\0';
 	}
 
-	for (auto& row = results.begin(); row != results.end(); ++row) {
-		i = Strings::ToInt(row[0]);
-		const EQ::ItemData *item_data = database.GetItem(Strings::ToInt(row[1]));
-		if (!item_data)
+	for (const auto& e : l) {
+		const auto* item_data = database.GetItem(e.item_id);
+		if (!item_data) {
 			continue;
-		pp->potionbelt.Items[i].ID = item_data->ID;
-		pp->potionbelt.Items[i].Icon = Strings::ToInt(row[2]);
-		strncpy(pp->potionbelt.Items[i].Name, item_data->Name, 64);
+		}
+
+		pp->potionbelt.Items[e.potion_id].ID   = item_data->ID;
+		pp->potionbelt.Items[e.potion_id].Icon = e.icon;
+
+		strn0cpy(pp->potionbelt.Items[e.potion_id].Name, item_data->Name, sizeof(pp->potionbelt.Items[e.potion_id].Name));
```

Comment:
I don't think its smart of us to use the size of something that isn't set yet in `sizeof`